### PR TITLE
chore(HOME-3506): v10 external transfer raml

### PR DESCRIPTION
**Pull request by kongmalin**
Closes: HOME-3506

### CODE REVIEW

**Comment by kongmalin** [comment]
screen shot 2018-05-21 at 4 23 46 pm
screen shot 2018-05-21 at 4 24 01 pm
screen shot 2018-05-21 at 4 24 22 pm
screen shot 2018-05-21 at 4 24 36 pm
screen shot 2018-05-21 at 4 24 52 pm
screen shot 2018-05-21 at 4 25 02 pm

File: api/v10/schema/moneyTransfer.res.json [hunk 1]
```diff
@@ -0,0 +1,131 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "description": "Money Transfer",
+  "properties": {
+    "transactionId": {
+      "description": "Veo money transfer id",
```
Comment (bondricek) [inline api/v10/schema/moneyTransfer.res.json]: remove reference to veo

File: api/v10/schema/moneyTransfer.res.json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        "PENDING",
+        "SUBMITTED",
+        "FAILED",
+        "COMPLETED",
+        "VEO_NEW",
+        "VEO_REJECTED",
+        "VEO_APPROVED",
+        "VEO_POST_DATED",
+        "VEO_RECURRING",
+        "VEO_SUCCESS",
```
Comment (bondricek) [inline api/v10/schema/moneyTransfer.res.json]: are we really returning veo statues here?

File: api/v10/schema/moneyTransfers.res.json [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      "items": { "$ref": "#/definitions/moneyTransfer" }
+    }
+  },
+  "definitions": {
+    "moneyTransfer": {
+      "type": "object",
+      "description": "Money Transfer",
+      "properties": {
+        "transactionId": {
+          "description": "Veo money transfer id",
```
Comment (bondricek) [inline api/v10/schema/moneyTransfers.res.json]: same replace `veo` with `custodian`

File: api/v10/schema/moneyTransfers.res.json [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+            "PENDING",
+            "SUBMITTED",
+            "FAILED",
+            "COMPLETED",
+            "VEO_NEW",
+            "VEO_REJECTED",
+            "VEO_APPROVED",
+            "VEO_POST_DATED",
+            "VEO_RECURRING",
+            "VEO_SUCCESS",
```
Comment (bondricek) [inline api/v10/schema/moneyTransfers.res.json]: same question, are we really returning veo statuses?

File: api/v10/schema/moneyTransfer.res.json (hunk 2 above)
Comment (kongmalin) [reply api/v10/schema/moneyTransfer.res.json]: Since it is TransferStatus which contains them, so should we take them out?

**Comment by bondricek** [comment]
the mock is returning 200 instead of 204 for the delete external transfer
screen shot 2018-05-23 at 12 19 13 pm

**Comment by zrangerz** [comment]
Thanks for catching that!

Let me re-iterate: the mocks/docs for the RAML/API are CRITICAL now. They need to be tested like prod code. We need to test and fail if we return the wrong status code.

—
Steve Mays
www.mrmays.com

**Comment by kongmalin** [comment]
Thank Brad. I am trying to replicate your test in QA, but it does not allow since there is no request body.
screen shot 2018-05-23 at 2 08 33 pm

But the error message contain the dto.

**Comment by bondricek** [comment]
sorry the screen shot did not capture but i did include the following as
the request body with type application/json
```
{
  "transactionId":"99999999",
  "transferType":"DEPOSIT"
}
```

**Comment by bondricek** [comment]
we need to fix the dto in the error message, separate issue

**Pull request merged**